drivers/r2dbc-driver/src/main/kotlin/app/cash/sqldelight/driver/r2dbc/R2dbcDriver.kt

```diff
@@ -13,15 +13,47 @@ import kotlinx.coroutines.reactive.asFlow
 import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactive.awaitSingle
 
-class R2dbcDriver(private val connection: Connection) : SqlDriver {
+class R2dbcDriver(private val connection: Connection, private val replaceParameter: Boolean) : SqlDriver {
+  private fun calcParameterReplacementAdditionalLength(parameters: Int): Int {
+    var numbers = 9 // numbers with current length
```

Comment:
Could you please explain this code?

Reply:
I could add extensive comments to explain the code in depth:

```
  /**
   * Calculate the length required to replace [parameters] `?` with `$n`, starting at `$1`
   * This is equivalent to the length of the string all numbers from 1 to [parameters] concatenated would require.
   */
  private fun calcParameterReplacementAdditionalLength(parameters: Int): Int {
    var numbers = 9 // count of numbers with current length (1 to 9 in the first step -> 9)
    var strLen = 1 // length of each number in ascii characters (1 to 9 -> 1, 10-99 -> 2 and so on)
    var remaining = parameters // numbers we did not yet consider
    var lengthSum = 0 // length of all numbers concatenated up to now
    while (remaining > numbers) { // as long as we have more remaining numbers than numbers of length [strLen]
      remaining -= numbers // reduce the number of remaining numbers by the amount of numbers with the current length
      // every number considered in this step consists of strLen digits. Therefor a string of all of them concatenated 
      // would have length [strLen] * [numbers]
      lengthSum += strLen * numbers
      // there are 10 times as many numbers having with a represantation of one more digit:
      // 1 digit: 1-9 -> 9
      // 2 digits: 10-99 -> 90 (= 99 - 9)
      // 3 digits: 100-999 -> 900 (= 999 - 99)
      // ...
      numbers *= 10
      strLen += 1
    }
    // there are less than or exactly [numbers] numbers left with a string represantation of at most [strLen] digits 
    // -> just add them to the already counted string length
    return lengthSum + remaining * strLen
  }
```